### ISSUE TITLE
vue/component-name-in-template-casing `ignores` should allow wildcards

### ISSUE THREAD

**Issue by ffxsam** [vuejs/eslint-plugin-vue#721]
Currently, the `vue/component-name-in-template-casing` rule has an `ignores` parameter to allow specific component names to bypass whatever rule the developer has set. It would be even more useful to allow wildcards, so we could avoid applying Pascal case to third party UI libraries, e.g.:

```js
    'vue/component-name-in-template-casing': [
      'error',
      'PascalCase',
      {
        ignores: ['el-*'],
      },
    ],
```

So when using Element UI's components (e.g. `<el-button>`), those will no longer be shown as errors if they're not in Pascal case.

**Comment by armano2** [comment]
https://github.com/vuejs/eslint-plugin-vue/pull/714
